packages/database/src/utils/audit/attachChangelogToSnapshotRecords.ts

```diff
@@ -0,0 +1,57 @@
+import { QueryTypes, type Sequelize } from 'sequelize';
+
+import type {
+  ChangelogRecord,
+  SyncSnapshotAttributes,
+  SyncSnapshotAttributesWithChangelog,
+} from 'types/sync';
+
+type QueryConfig = {
+  minSourceTick: number;
+  maxSourceTick?: number;
+  tableWhitelist?: string[];
+};
+
+export const attachChangelogToSnapshotRecords = async (
+  sequelize: Sequelize,
+  snapshotRecords: SyncSnapshotAttributes[],
+  { minSourceTick, maxSourceTick, tableWhitelist }: QueryConfig,
+): Promise<SyncSnapshotAttributesWithChangelog[]> => {
+  if (!snapshotRecords.length) {
+    return snapshotRecords;
+  }
+  const changelogRecords = (await sequelize.query(
+    `
+    SELECT * FROM logs.changes
+    WHERE updated_at_sync_tick > :minSourceTick
+    ${maxSourceTick ? 'AND updated_at_sync_tick < :maxSourceTick' : ''}
+    ${tableWhitelist ? `AND table_name IN (:tableWhitelist)` : ''}
```

Comment:
We don't technically need a whitelist here if we can filter the `recordTypeAndIds` array instead, but I also think it keeps things clear and probably does no harm, maybe even a little good to the query performance